Services/Mentor/OJS.Services.Mentor.Business/Implementations/MentorBusinessService.cs

```diff
@@ -0,0 +1,480 @@
+
+                    // Case 2: Check if we should start collecting from this section based on problem number
+                    if (sectionCount == problemNumber)
+                    {
+                        shouldCollectContent = true;
+                        resultContent.AppendLine($"## {text}");
+                    }
+                    else if (sectionCount > problemNumber)
+                    {
+                        resultContent.AppendLine($"## {text}");
+                    }
+
+                    currentHeading = text;
+                }
+                else
+                {
+                    // For Case 3, collect everything
+                    if (problemNumber > sectionCount)
+                    {
+                        resultContent.AppendLine(text);
+                    }
+                    // For other cases, collect only when flagged
+                    else if (shouldCollectContent)
+                    {
+                        resultContent.AppendLine(text);
+                    }
+                }
+            }
+            else if (element is Table table)
+            {
+                // For Case 3, collect all tables
+                if (problemNumber > sectionCount)
+                {
+                    ProcessTable(table, resultContent);
+                }
+                // For other cases, collect only when flagged
+                else if (shouldCollectContent)
+                {
+                    ProcessTable(table, resultContent);
+                }
+            }
+        }
+
+        return resultContent.ToString().Trim();
+    }
+
+    private static void ProcessTable(Table table, StringBuilder matchingSectionContent)
+    {
+        var rows = table.Elements<TableRow>().ToList();
+        if (rows.Count == 0)
+        {
+            return;
+        }
+
+        /*
+         * Currently, this implementation accepts that the first row will be of headers,
+         * this might need to be generalized after testing.
+         */
+        var headerRow = rows.First();
+        var headers = headerRow.Elements<TableCell>()
+            .Select(cell => cell.InnerText.Trim())
+            .ToList();
+
+        foreach (var row in rows.Skip(1))
+        {
+            var cells = row.Elements<TableCell>().ToList();
+            var rowValues = cells.Select(cell =>
+            {
+                var runs = cell.Descendants<Text>();
+                return string.Join(" ", runs.Select(r => r.Text.Trim()));
+            }).ToList();
+
+            if (rowValues.All(string.IsNullOrEmpty))
+            {
+                continue;
+            }
+
+            var rowData = new StringBuilder();
+            for (var i = 0; i < Math.Min(headers.Count, rowValues.Count); i++)
+            {
+                if (!string.IsNullOrEmpty(rowValues[i]))
+                {
+                    if (rowData.Length > 0)
+                    {
+                        rowData.Append(" ⦚ ");
```

Comment:
What is this symbol? Do we use it somewhere else? Maybe extract into a constant.